ide/languages.hcl/test/unit/src/org/netbeans/modules/languages/hcl/HCLIndenterTest.java

```diff
@@ -118,7 +118,18 @@ public void testEmpty4() throws Exception {
 
     @Test
     public void testEmptyLine() throws Exception {
-        performLineIndentationTest("locals {\n  |\n}\n", "locals {\n\n}\n");
+        performLineIndentationTest(
+                """
+                locals {
+                __|
```

Comment:
Is the special casing  necessary? The pipe symbol to mark the place of the cursor I knew, the underscores look out of place. In fact I felt obligated to read the text block JEP again to see whether I missed something.

Reply:
This is a TextBlock experiment in formatting. Yes `|` stands for the cursor and `_` would stand for "visible space".
the `performLineIndentationTest` will turn them to spaces. I'm not entirely convinced yet, but a bit easier to read the strings this way, than decoding `\n` and ` ` sequences.

Reply:
My point was only, that I would drop the extra `_` symbols.